### ISSUE TITLE
Matrix multiplication between Mat and Matx

### ISSUE THREAD

**Issue by catree** [opencv/opencv#13926]
There is an issue when doing a matrix multiplication between a `Mat` and a `Matx`.

Code:

```
#include <opencv2/opencv.hpp>
using namespace cv;

int main()
{
    Mat M1 = (Mat_<double>(4,4,CV_64FC1) << 1, 2, 3, 4,
                                            5, 6, 7, 8,
                                            9, 10, 11, 12,
                                            13, 14, 15, 16);

    Matx44d M2(1, 2, 3, 4,
               5, 6, 7, 8,
               9, 10, 11, 12,
               13, 14, 15, 16);

    std::cout << "M1*M1:\n" << M1*M1 << std::endl; //OK
    std::cout << "M2*M2:\n" << M2*M2 << std::endl; //OK
    std::cout << "M1*M2:\n" << M1*M2 << std::endl; //NOK
    std::cout << "M2*M1:\n" << M2*M1 << std::endl; //NOK
    std::cout << "M1*Mat(M2):\n" << M1*Mat(M2) << std::endl; //OK

    return 0;
}
```

Output:

```
M1*M1:
[90, 100, 110, 120;
 202, 228, 254, 280;
 314, 356, 398, 440;
 426, 484, 542, 600]
M2*M2:
[90, 100, 110, 120;
 202, 228, 254, 280;
 314, 356, 398, 440;
 426, 484, 542, 600]
M1*M2:
[2, 4, 6, 8;
 10, 12, 14, 16;
 18, 20, 22, 24;
 26, 28, 30, 32]
M2*M1:
[2, 4, 6, 8;
 10, 12, 14, 16;
 18, 20, 22, 24;
 26, 28, 30, 32]
M1*Mat(M2):
[90, 100, 110, 120;
 202, 228, 254, 280;
 314, 356, 398, 440;
 426, 484, 542, 600]
```

**Comment by berak** [comment]
simple typo (in both master and 3.4):

https://github.com/opencv/opencv/blob/c3cf35ab63c04fb1d7b2f6760128f42c20cac0e1/modules/core/include/opencv2/core/mat.hpp#L3592-L3594

**Comment by terfendail** [comment]
Fixed by #13931
